From 327ed93b5a9a505d848992003e9d2aaf666e95c7 Mon Sep 17 00:00:00 2001
From: Devang Vyas <devangnayanbhai.vyas@amd.com>
Date: Fri, 7 Apr 2023 17:03:16 +0530
Subject: [PATCH 02/31] net: phy: aquantia: Add 10mbps support

This adds support for 10mbps speed in PHY device's
"supported" field which helps in autonegotiating
10mbps link from PHY side where PHY supports the speed
but not updated in PHY kernel framework.

One such example is AQR113C PHY.

Signed-off-by: Devang Vyas <devangnayanbhai.vyas@amd.com>
Change-Id: Idad2cd3a4e599016a1bed4cf998a3d4b5303b148
---
 drivers/net/phy/aquantia_main.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/phy/aquantia_main.c b/drivers/net/phy/aquantia_main.c
index 47a76df36b74..0e5512a5f0c7 100644
--- a/drivers/net/phy/aquantia_main.c
+++ b/drivers/net/phy/aquantia_main.c
@@ -554,6 +554,9 @@ static void aqr107_chip_info(struct phy_device *phydev)
 	build_id = FIELD_GET(VEND1_GLOBAL_RSVD_STAT1_FW_BUILD_ID, val);
 	prov_id = FIELD_GET(VEND1_GLOBAL_RSVD_STAT1_PROV_ID, val);
 
+	if (!test_bit(ETHTOOL_LINK_MODE_10baseT_Full_BIT, phydev->supported))
+		linkmode_set_bit(ETHTOOL_LINK_MODE_10baseT_Full_BIT, phydev->supported);
+
 	phydev_dbg(phydev, "FW %u.%u, Build %u, Provisioning %u\n",
 		   fw_major, fw_minor, build_id, prov_id);
 }
-- 
2.34.1

